syntax = "proto2";
package proto;
enum ReportId {
  option allow_alias = true;
  ReportIdGamepad = 1;
  ReportIdKeyboard = 2;
  ReportIdMouse = 3;
  ReportIdConsumer = 4;
  ReportIdConfig = 0x22;
  ReportIdConfigInfo = 0x23;
  ReportIdLoaded = 0x24;
  ReportIdKeepalive = 0x25;
  ReportIdBootloader = 0x26;
  ReportIdPs301 = 0x01;
  ReportIdPs302 = 0x02;
  ReportIdPs3EE = 0xEE;
  ReportIdPs3EF = 0xEF;
  ReportIdPs3F2 = 0xF2;
  ReportIdPs3F4 = 0xF4;
  ReportIdPs3F5 = 0xF5;
  ReportIdPs3F6 = 0xF6;
  ReportIdPs3F7 = 0xF7;
  ReportIdPs3F8 = 0xF8;
  ReportIdPs4Feature = 0x03;
  ReportIdPs405 = 0x05;
  ReportIdPs4SetChallenge = 0xF0;
  ReportIdPs4GetResponse = 0xF1;
  ReportIdPs4GetAuthStatus = 0xF2;
  ReportIdPs4GetAuthPageSize = 0xF3;
}

enum MappingMode {
  PerInput = 1; /* Have one set of mappings that covers all extensions */
  PerExtension = 2; /* Have individual mappings for each input in each extension */
}

enum FaceButtonMappingMode {
  LegendBased = 1;   /* Map face buttons by legend */
  PositionBased = 2; /* Map face buttons by position */
}

enum WiiExtType {
    WiiNunchuk = 0x0000;
    WiiClassicController = 0x0001;
    WiiClassicControllerPro = 0x0101;
    WiiThqUdrawTablet = 0xFF12;
    WiiUbisoftDrawsomeTablet = 0xFF13;
    WiiGuitarHeroGuitar = 0x0003;
    WiiGuitarHeroDrums = 0x0103;
    WiiDjHeroTurntable = 0x0303;
    WiiTaikoNoTatsujinController = 0x0011;
    WiiMotionPlus = 0x0005;
    WiiNoExtension = 0x180b;
    WiiNotInitialised = 0xFFFF;
} ;

enum SubType {
  Gamepad = 1;
  Dancepad = 2;
  GuitarHeroGuitar = 3;
  RockBandGuitar = 4;
  GuitarHeroDrums = 5;
  RockBandDrums = 6;
  LiveGuitar = 7;
  DjHeroTurntable = 8;
  ProGuitarMustang = 9;
  ProGuitarSquire = 10;
  ProKeys = 11;
  Taiko = 12;
  StageKit = 13;
  KeyboardMouse = 14;
  Wheel = 15;
  DisneyInfinity = 16;
  Skylanders = 17;
  LegoDimensions = 18;
}

enum PinMode {
  PullDown = 1;
  PullUp = 2;
  Floating = 3;
  BusKeep = 4;
}

enum Mpr121PinMode {
  Touch = 1;
  Digital = 2;
}

enum UsbDeviceType {
  Generic = 1;
  Santroller = 2;
  Xbox360 = 3;
  Xbox360Wireless = 4;
  Xbox360BigButton = 5;
  XboxOne = 6;
  XboxOG = 7;
  PS3 = 8;
  PS4 = 9;
  PS5 = 10;
  WiiRockBand = 11;
  Switch = 12;
  Raphnet = 13;
  StreamDeck = 14;
  LTEK = 15;
  Stadia = 16;
  Midi = 17;
  Keyboard = 18;
  Mouse = 19;
}
